pkg/master/reconcilers/lease.go

```diff
@@ -249,3 +261,11 @@ func checkEndpointSubsetFormatWithLease(e *api.Endpoints, expectedIPs []string,
 
 	return true, ipsCorrect, portsCorrect
 }
+
+func (r *leaseEndpointReconciler) StopReconciling(serviceName string, ip net.IP, endpointPorts []api.EndpointPort) error {
+	if err := r.masterLeases.RemoveLease(ip.String()); err != nil {
```

Comment:
any need for locking? r.masterLeases is thead safe? Also doReconcile?

Reply:
I don't think a lock is needed, but probably a control mechanism to prevent an update after StopReconciling has been called.

Prefer a different name instead of doReconcile?

Reply:
was just wondering as the shutdown is certainly a different go routine than the ongoing reconcilation.